docs/contributing.rst

```diff
@@ -159,34 +159,68 @@ Any extra texts (print statements and so on) should be removed.
      make test-3.10-no-extensions
 
 Tests coverage
---------------
+==============
 
-We are trying hard to have good test coverage; please don't make it worse.
+We use *codecov.io* as an indispensable tool for analyzing our coverage
+results. Visit https://codecov.io/gh/aio-libs/aiohttp to see coverage
+reports for the master branch, history, pull requests etc.
 
-Use:
+We'll use an example from a real PR to demonstrate how we use this.
+Once the tests run in a PR, you'll see a comment posted by *codecov*.
+The most important thing to check here is whether there are any new
+missed or partial lines in the report:
 
-.. code-block:: shell
+.. image:: _static/img/contributing-cov-comment.svg
+
+Here, the PR has introduced 1 miss and 2 partials. Now we
+click the link in the comment header to open the full report:
+
+.. image:: _static/img/contributing-cov-header.svg
+   :alt: Codecov report
+
+Now, if we look through the diff under 'Files changed' we find one of
+our partials:
 
-   $ make cov-dev
+.. image:: contributing-cov-partial.png
+   :alt: A while loop with partial coverage.
 
-to run test suite and collect coverage information. Once the command
-has finished check your coverage at the file that appears in the last
-line of the output:
-``open file:///.../aiohttp/htmlcov/index.html``
+In this case, the while loop is never skipped in our tests. This is
+probably not worth writing a test for (and may be a situation that is
+impossible to trigger anyway), so we leave this alone.
 
-Please go to the link and make sure that your code change is covered.
+We're still missing a partial and a miss, so we switch to the
+'Indirect changes' tab and take a look through the diff there. This
+time we find the remaining 2 lines:
 
+.. image:: contributing-cov-miss.png
+   :alt: An if statement that isn't covered anymore.
 
-The project uses *codecov.io* for storing coverage results. Visit
-https://codecov.io/gh/aio-libs/aiohttp for looking on coverage of
-master branch, history, pull requests etc.
+After reviewing the PR, we find that this code is no longer needed as
+the changes mean that this method will never be called under those
+conditions. Thanks to this report, we were able to remove some
+redundant code from a performance-critical part of our codebase (this
+check would have been run, probably multiple times, for every single
+incoming request).
+
```

Comment:
Can we add a `.. tip::` block somewhere here mentioning that Codecov is sometimes drunk and the metrics reported may be incomplete at times. It might be a good transition paragraph to hint the contributor to keep reading and point them towards the importance of inspecting the coverage locally as well.